### PR TITLE
Prevent empty header arrays

### LINKED ISSUE #230: Prevent empty header arrays
`Message::setHeader()` shouldn't create an array entry if the header array is empty, but instead delete the current entry if it exists. Similar for `Message::addHeader()`. There should also be a `array_values()` call to ensure no string keys are present in the array.

### PR DESCRIPTION
**Pull request by ekinhbayar**
Fixes #230

### CODE REVIEW

**Comment by kelunik** [comment]
@ekinhbayar Do you want to add a test for the interaction of `setHeader` and `hasHeader`?

**Comment by kelunik** [comment]
Thanks!

File: src/Message.php [hunk 1]
```diff
@@ -86,7 +91,12 @@ public function addHeader(string $name, $value) {
         \assert($this->isNameValid($name), "Invalid header name");
 
         if (\is_array($value)) {
-            $value = \array_map("strval", $value);
+            if (!$value) {
+                $this->removeHeader($name);
```
Comment (trowski) [inline src/Message.php]: The header should not be removed from this method, this should just `return`.

Comment (kelunik) [reply src/Message.php]: Oops, yes. Missed that.

Comment (kelunik) [reply src/Message.php]: Fixed via acb8019923f4e8fed6abcff1fdf2dae9849f5c54.

**Pull request merged**